src/util/md5.cpp

```diff
@@ -1,14 +1,263 @@
+#include <fstream>
+#include <iomanip>
+#include <iostream>
+#include <sstream>
+#include <string>
 #include "md5.h"
+using namespace std;
 
-/**
- * Initializes the MD5 hash. Don't forget to call hashMd5() afterwards to actually get your hash
- */
-void md5_reset(md5hash_t& hash) {
-    // seed the hash
-    hash.dwords[0] = 0x67452301;
-    hash.dwords[1] = 0xefcdab89;
-    hash.dwords[2] = 0x98badcfe;
-    hash.dwords[3] = 0x10325476;
+
+// Constants for MD5Transform routine
+#define S11 7
+#define S12 12
+#define S13 17
+#define S14 22
+#define S21 5
+#define S22 9
+#define S23 14
+#define S24 20
+#define S31 4
+#define S32 11
+#define S33 16
+#define S34 23
+#define S41 6
+#define S42 10
+#define S43 15
+#define S44 21
+
+// Function Prototypes
+void MD5Transform(uint32_t state[4], const unsigned char block[64]);
+void Encode(unsigned char *output, const uint32_t *input, unsigned int len);
+void Decode(uint32_t *output, const unsigned char *input, unsigned int len);
+void MD5_memcpy(unsigned char *output, const unsigned char *input, unsigned int len);
+void MD5_memset(unsigned char *output, int value, unsigned int len);
+
+// F, G, H and I are basic MD5 functions
+#define F(x, y, z) (((x) & (y)) | ((~x) & (z)))
+#define G(x, y, z) (((x) & (z)) | ((y) & (~z)))
+#define H(x, y, z) ((x) ^ (y) ^ (z))
+#define I(x, y, z) ((y) ^ ((x) | (~z)))
+
+// ROTATE_LEFT rotates x left n bits
+#define ROTATE_LEFT(x, n) (((x) << (n)) | ((x) >> (32-(n))))
+
+// FF, GG, HH, and II transformations
+#define FF(a, b, c, d, x, s, ac) { \
+ (a) += F ((b), (c), (d)) + (x) + (uint32_t)(ac); \
+ (a) = ROTATE_LEFT ((a), (s)); \
+ (a) += (b); \
+  }
+#define GG(a, b, c, d, x, s, ac) { \
+ (a) += G ((b), (c), (d)) + (x) + (uint32_t)(ac); \
+ (a) = ROTATE_LEFT ((a), (s)); \
+ (a) += (b); \
+  }
+#define HH(a, b, c, d, x, s, ac) { \
+ (a) += H ((b), (c), (d)) + (x) + (uint32_t)(ac); \
+ (a) = ROTATE_LEFT ((a), (s)); \
+ (a) += (b); \
+  }
+#define II(a, b, c, d, x, s, ac) { \
+ (a) += I ((b), (c), (d)) + (x) + (uint32_t)(ac); \
+ (a) = ROTATE_LEFT ((a), (s)); \
+ (a) += (b); \
+  }
+
+// MD5 initialization
+void MD5Init(MD5_CTX_t *context) {
```

Comment:
We should be consistent in name style/convention; probably avoid function names that start with upper-case, and be consistent in use of underscores?

Since this is all new code, we should also use Doxygen blocks for each function.